internal/exec/engine.go

```diff
@@ -189,28 +191,49 @@ func (e Engine) fetchProviderConfig(f resource.Fetcher) (types.Config, error) {
 // set, the referenced and evaluted config will be returned. Otherwise, if
 // "ignition.config.append" is set, each of the referenced configs will be
 // evaluated and appended to the provided config. If neither option is set, the
-// provided config will be returned unmodified.
-func (e *Engine) renderConfig(cfg types.Config, f resource.Fetcher) (types.Config, error) {
+// provided config will be returned unmodified. An updated fetcher will be
+// returned with any new timeouts set.
+func (e *Engine) renderConfig(cfg types.Config, f resource.Fetcher) (types.Config, resource.Fetcher, error) {
 	// Apply any new timeout info before fetching other configs.
 	e.client = resource.NewHttpClient(e.Logger, cfg.Ignition.Timeouts)
 	if cfgRef := cfg.Ignition.Config.Replace; cfgRef != nil {
-		return e.fetchReferencedConfig(*cfgRef, f)
+		newCfg, err := e.fetchReferencedConfig(*cfgRef, f)
+
+		// Replace the HTTP client in the fetcher to be configured with the
+		// timeouts of the new config
+		client := resource.NewHttpClient(e.Logger, newCfg.Ignition.Timeouts)
+		f.Client = &client
+
+		newCfg, f, err = e.renderConfig(newCfg, f)
+		if err != nil {
+			return newCfg, f, err
+		}
+		return newCfg, f, err
 	}
 
 	appendedCfg := cfg
 	for _, cfgRef := range cfg.Ignition.Config.Append {
 		newCfg, err := e.fetchReferencedConfig(cfgRef, f)
 		if err != nil {
-			return newCfg, err
+			return newCfg, f, err
+		}
+
+		// Replace the HTTP client in the fetcher to be configured with any new
+		// timeouts so that they're in effect when fetching the next config
+		client := resource.NewHttpClient(e.Logger, config.Append(appendedCfg, newCfg).Ignition.Timeouts)
+		f.Client = &client
+
+		newCfg, f, err = e.renderConfig(newCfg, f)
+		if err != nil {
+			return newCfg, f, err
```

Comment:
Same thing here with returning an empty config, nil and err.

Reply:
If a caller of `renderConfig` does something like `cfg, f, err := e.renderConfig(cfg, f)` and it fails, I think it makes sense to keep the fetcher functional instead of replacing it with a `resource.Fetcher{}` (can't be `nil` because it's not a pointer), just in case there's a scenario in the future where we want to do network access on failure. Callers should check the results of `err` before proceeding anyway.

I'll change this to return `types.Config{}` though.